lib/workload_mapper.rb

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) 2013-2015 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of version 3 of the GNU General Public License as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE about this file by physical or electronic mail,
+# you may find current contact information at www.suse.com
+class WorkloadMapper
+  attr_reader :workloads, :current_workload
+
+  def initialize(&block)
+    @workloads = {}
```

Comment:
Is there any reason to keep the workloads as a member variable. Returning it in `identify_workloads` and using it as an argument for `write_compose_file` would minimize local state. This would make `WorkloadMapper` more flexible and make the code easier to read as there are less implicit assumptions.